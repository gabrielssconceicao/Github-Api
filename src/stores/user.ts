import { ref } from 'vue';
import { defineStore } from 'pinia';
import type { IUserResponse } from '@/types/user';
import { fetchUser } from '@/hooks/fetchUser';
export const useUserStore = defineStore('counter', () => {
  const githubUser = ref<IUserResponse>({} as IUserResponse);

  async function searchUser(userLogin: string) {
    const data = await fetchUser(userLogin);
    githubUser.value = data;
  }

  return { githubUser, searchUser };
});
