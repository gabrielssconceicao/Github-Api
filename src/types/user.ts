export interface IUser {
  name: null | string;
  login: string;
  avatar_url: string;
  followers: number;
  following: number;
  location: null | string;
  html_url: string;
  bio: null | string;
  public_repos: number;
  repos_url: string;
}

export interface IUserResponse {
  user: IUser;
  status: number;
}
