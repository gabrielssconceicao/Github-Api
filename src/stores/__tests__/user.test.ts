import { describe, it, expect } from 'vitest';
import { useUserStore } from '../user';
import { userResponseError, userResponseSuccess } from '@/mock/__tests__/user';
describe('userStore', () => {
  it('should init with an empty user', () => {
    const userStore = useUserStore();
    expect(userStore.githubUser.status).toBe(undefined);
    expect(userStore.githubUser.user).toBe(undefined);
    expect(userStore.inputValue).toBe('');
    expect(userStore.userName).toBe(undefined);
    const { githubUser, inputValue, userName } = userStore;
    expect({ githubUser, inputValue, userName }).toMatchSnapshot();
  });

  it('should set input value', () => {
    const userStore = useUserStore();
    userStore.setInputValue('teste');
    expect(userStore.inputValue).toBe('teste');
    expect(userStore.githubUser.status).toBe(undefined);
    expect(userStore.githubUser.user).toBe(undefined);
    expect(userStore.userName).toBe(undefined);
    const { githubUser, inputValue, userName } = userStore;
    expect({ githubUser, inputValue, userName }).toMatchSnapshot();
  });

  it('should search user ', async () => {
    const userStore = useUserStore();
    userStore.setInputValue('resolveTest');
    await userStore.searchUser();
    expect(userStore.githubUser.status).toBe(userResponseSuccess.status);
    expect(userStore.githubUser.user).toEqual(userResponseSuccess.user);
    expect(userStore.userName).toBe('Teste User');
    const { githubUser, inputValue, userName } = userStore;
    expect({ githubUser, inputValue, userName }).toMatchSnapshot();
  });

  it('should search user error', async () => {
    const userStore = useUserStore();
    userStore.setInputValue('rejectTest');
    await userStore.searchUser();
    expect(userStore.githubUser.status).toBe(userResponseError.status);
    expect(userStore.githubUser.user).toEqual(userResponseError.user);
    const { githubUser, inputValue } = userStore;
    expect({ githubUser, inputValue }).toMatchSnapshot();
  });
});
