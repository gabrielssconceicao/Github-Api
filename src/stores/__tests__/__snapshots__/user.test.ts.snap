// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`userStore > should init with an empty user 1`] = `
{
  "githubUser": {},
  "inputValue": "",
  "userName": undefined,
}
`;

exports[`userStore > should search user  1`] = `
{
  "githubUser": {
    "status": 200,
    "user": {
      "avatar_url": "https://avatars.githubusercontent.com/u/10590076?v=4",
      "bio": "lorem ipsum dolor sit amet",
      "followers": 0,
      "following": 5,
      "html_url": "https://github.com/testeuser",
      "location": "Brazil",
      "login": "testeuser",
      "name": "Teste User",
      "public_repos": 5,
      "repos_url": "https://api.github.com/users/testeuser/repos",
    },
  },
  "inputValue": "resolveTest",
  "userName": "Teste User",
}
`;

exports[`userStore > should search user error 1`] = `
{
  "githubUser": {
    "status": 404,
    "user": {},
  },
  "inputValue": "rejectTest",
}
`;

exports[`userStore > should set input value 1`] = `
{
  "githubUser": {},
  "inputValue": "teste",
  "userName": undefined,
}
`;
